File: LeetCode/Medium/0189-rotate-array/0189-rotate-array.java
class Solution {
    public void rotate(int[] nums, int k) {
        int n = nums.length;
        int[] rottated = new int[n];

        for (int i = 0; i < n; i++) {
            rottated[(i+k)%n] = nums[i];
        }

        for (int i = 0; i < n; i++) {
            nums[i] = rottated[i];
        }
    }
}

File: LeetCode/Medium/0189-rotate-array/0189-rotate-array.go
func rotate(nums []int, k int)  {
    n := len(nums)
    copied := make([]int, n)
    copy(copied, nums)

    for i := 0; i < n; i++ {
        nums[(i+k)% n] = copied[i]
    }
    
}
 